src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+          Collection: *Collection
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {$match: {$or: {^Array: {of: 
+                  {$and: [
+                    {a: {$gt: {^Inc: {start: 10}}}},
+                    {b: {$gt: {^Inc: {start: 10}}}}
+                  ]},
+                  number: {^Parameter: {Name: "NumberOfClauses", Default: -1}}
+                }}}}
+              ]
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Collection: *Collection
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: *DocumentCount
+        Document:
+          a: [0, 1]
+          b: [2, 3]
+          c: 4
+          d: 5
+          object: {}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50Clauses
+      Repeat: 1000
+      ActivePhase: 3
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith100Clauses
+      Repeat: 1000
+      ActivePhase: 4
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith1000Clauses
+      Repeat: 100
+      ActivePhase: 5
+      NumberOfClauses: 1000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith10000Clauses
+      Repeat: 10
+      ActivePhase: 6
+      NumberOfClauses: 10000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50000Clauses
+      Repeat: 5
+      ActivePhase: 7
+      NumberOfClauses: 50000
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone
+      - standalone-all-feature-flags
+      - standalone-classic-query-engine
+      - standalone-heuristic-bonsai
+      - standalone-sampling-bonsai
+      - standalone-sbe
+      - standalone-query-stats
+    branch_name:
+      $gte:
+        v7.0
```

Comment:
You definitely want to set the $gte to the targeted version of your ticket- which will probably go in for 7.2, since we branched 7.1 already (unless this change will be backported?)

Reply:
I don't think it will be backported, set it to 7.2